File: Java/Week_2/Day4ExceptionHandling.java
package Week_2;

import java.util.Scanner;

public class Day4ExceptionHandling {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        
        // Basic try-catch
        try {
            System.out.print("Enter a number to deivide 10: ");
            int num = sc.nextInt();
            int result = 10 / num;
            System.out.println("Result: " + result);
        } catch (ArithmeticException e) {
            System.out.println("Can't divide by zero!");
        } catch (Exception e) {
            System.out.println("Something went wrong: " + e.getMessage());
        }
        
        // try-catch-finally
        try {
            System.out.print("Enter your age: ");
            int age = sc.nextInt();
            System.out.println("You entered: " + age);
        } catch (Exception e) {
            System.out.println("Invalid input, must be a number");
        } finally {
            System.out.println("This always runs, no matter what");
        }
        
        sc.close();
    }
}

File: Python/Week_2/day4_exception_handling.py
# Week 2 - Day 4 - Exception Handling in Python

# Basic try-except
try:
    num = int(input("Enter a number to devide 10: "))
    result = 10 / num
    print("Result:", result)
except ValueError:
    print("That's not a valid number!")
except ZeroDivisionError:
    print("Can't divide by zero!")

# try-except-else-finally
try:
    age = int(input("Enter your age: "))
except ValueError:
    print("Invalid input, must be a number")
else:
    print("You entered:", age)   # runs only if no exception happened
finally:
    print("This always runs, no matter what")   # cleanup code

# Catching any exception generically
try:
    x = 10 / 0
except Exception as e:
    print("Something went wrong:", e)